### ISSUE TITLE
TypeError: Cannot read property '$channel' of undefined

### ISSUE THREAD

**Issue by long1eu** [grpc/grpc-node#1331]
I get this error when I try to use util.promisify.

```
TypeError: Cannot read property '$channel' of undefined
    at Client.makeUnaryRequest (node_modules/grpc/src/client.js:547:19)
    at method_func (node_modules/grpc/src/client.js:1000:43)
    at internal/util.js:297:30
    at new Promise (<anonymous>)
    at method_func (internal/util.js:296:12)
    at Context.<anonymous> (test/controllers/tags.test.ts:39:79)
    at processImmediate (internal/timers.js:456:21)
```

if I don't use promisify then the call work perfectly 

```
import { MyClient } from "../../src/proto/my_grpc_pb";
import * as grpc from "grpc";
import { StringValue } from "google-protobuf/google/protobuf/wrappers_pb";
import * as util from "util";

const assert = require("chai").assert;

describe("tags", () => {
  const client = new MyClient(
    "localhost:8080",
    grpc.credentials.createInsecure()
  );

  it("should do this", async () => {
    const request = new StringValue();
    request.setValue("coffee");

    const result = await util.promisify(client.createTag)(request);
    console.log(`result ${result}`);
    assert.isUndefined(result);
  });
});
```
I'm using node v12.16.1 and grpc 1.24.2 on macOS 10.15.2.

**Comment by murgatroid99** [comment]
You can't just promisify a method from a class instance. You need to bind the method to the instance first.

**Comment by long1eu** [comment]
Do you mind pointing what I should change?

**Comment by murgatroid99** [comment]
I suggest reading the last part of the [`util.promisify`](https://nodejs.org/api/util.html#util_util_promisify_original) documentation.

**Comment by long1eu** [comment]
Thank you.

**Comment by udayrajMT** [comment]
TLDR; for upcoming visitors - 
```js
  const client = new MyClient(...);
  const createTag = util.promisify(client.createTag).bind(client);
  // ...
  const result = await createTag(request);
```

**Comment by Shahrukh-UXDesign** [comment]
**I have the same issue  but regarding the npm audit fix
-------------------------------------------------------------------------------**

npm audit fix
npm ERR! Cannot read property '@grpc/grpc-js' of undefined

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/wounderland/.npm/_logs/2023-01-23T10_10_19_539Z-debug.log
